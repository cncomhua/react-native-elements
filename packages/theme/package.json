{
  "name": "@react-native-elements/theme",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.tsx",
  "react-native": "src/index.tsx",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "peerDependencies": {
    "@react-native-element/base": "*"
  },
  "author": "",
  "license": "ISC"
}
