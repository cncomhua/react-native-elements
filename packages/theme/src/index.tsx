import React from "react";
import { Pressable, Text, StyleSheet } from "react-native";
// import { FAB } from "./src/FAB/FAB";

const s = StyleSheet.create({
  button: {
    backgroundColor: "#f00",
    padding: 10,
    margin: 10,
    borderRadius: 10,
  },
});
export const ButtonComp = () => {
  return (
    <>
      {/* <Tab value={0}>
        <TabItem title="Tab 1" />
        <TabItem title="Tab 2" />
        <TabItem title="Tab 3" />
      </Tab> */}
      {/* 
      <FAB title={"FAB"} />
      <Button title="working" /> */}
      <Pressable style={s.button}>
        <Text>Yes Working...</Text>
      </Pressable>
    </>
  );
};

// export * from "./src";
