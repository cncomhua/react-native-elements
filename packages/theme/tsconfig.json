{
  "extends": "../../tsconfig",
  "references": [{ "path": "../base" }],
  "compilerOptions": {
    "allowJs": true /* Allow javascript files to be compiled. */,
    "rootDir": "src",
    "outDir": "dist",
    "declaration": true,
    "composite": true
  },
  "include": [
    "src/**/*.tsx",
    "src/**/*.ts",
    "src/index.ts",
    "../base/src/**/*.tsx",
    "../base/src/**/*.ts",
    "../base/src/index.ts"
  ],
  "exclude": [
    ".ci",
    "dist",
    "node_modules",
    "src/index.d.ts",
    "src/**/__tests__",
    "babel.config.js",
    "metro.config.js",
    "jest.config.js",
    "website"
  ]
}
