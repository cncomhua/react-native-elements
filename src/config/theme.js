// For testing purposes only
import colors from './colors';

export default {
  colors,
};
