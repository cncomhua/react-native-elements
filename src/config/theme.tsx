import { ViewProps } from 'react-native';
import { SwipeRatingProps } from '../Rating';
import { TapRatingProps } from '../AirbnbRating';
import { AvatarProps, AccessoryProps } from '../Avatar';
import { BadgeProps } from '../Badge';
import { BottomSheetProps } from '../BottomSheet';
import { ButtonProps } from '../Button';
import { ButtonGroupProps } from '../ButtonGroup';
import { CardProps } from '../card/Card';
import { CheckBoxProps } from '../Checkbox';
import { DividerProps } from '../Divider';
import { HeaderProps } from '../Header';
import { IconProps } from '../Icon';
import { ImageProps } from '../Image';
import { InputProps } from '../Input';
import { ListItemProps } from '../ListItem';
import { ListItemAccordionProps } from '../ListItem';
import { OverlayProps } from '../Overlay';
import { PricingCardProps } from '../PricingCard';
import { SearchBarProps } from '../searchbar/SearchBar';
import { SliderProps } from '../Slider';
import { SocialIconProps } from '../SocialIcon';
import { TextProps } from '../Text';
import { TileProps } from '../Tile';
import { TooltipProps } from '../Tooltip';
import { SwitchProps } from '../Switch';
import { TabItemProps, TabProps } from '../tab/Tab';
import { TabViewProps } from '../tab/TabView';
import { FABProps } from '../FAB';
import { SpeedDialProps, SpeedDialActionProps } from '../SpeedDial';
import { LinearProgressProps } from '../LinearProgress';
import { ChipProps } from '../Chip';
import {
  DialogActionsProps,
  DialogLoadingProps,
  DialogProps,
  DialogTitleProps,
} from '../Dialog';

import colors, { Colors } from './colors';

export default {
  colors,
};

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface FullTheme {
  Avatar: Partial<AvatarProps>;
  AvatarAccessory: Partial<AccessoryProps>;
  Badge: Partial<BadgeProps>;
  BottomSheet: Partial<BottomSheetProps>;
  Button: Partial<ButtonProps>;
  ButtonGroup: Partial<ButtonGroupProps>;
  Chip: Partial<ChipProps>;
  Card: Partial<CardProps>;
  CardDivider: Partial<DividerProps>;
  CardFeaturedSubtitle: Partial<TextProps>;
  CardFeaturedTitle: Partial<TextProps>;
  CardImage: Partial<ImageProps>;
  CardTitle: Partial<TextProps>;
  CheckBox: Partial<CheckBoxProps>;
  Divider: Partial<DividerProps>;
  Dialog: Partial<DialogProps>;
  DialogActions: Partial<DialogActionsProps>;
  DialogButton: Partial<ButtonProps>;
  DialogLoading: Partial<DialogLoadingProps>;
  DialogTitle: Partial<DialogTitleProps>;
  Header: Partial<HeaderProps>;
  Icon: Partial<IconProps>;
  Image: Partial<ImageProps>;
  Input: Partial<InputProps>;
  ListItem: Partial<ListItemProps>;
  ListItemAccordion: Partial<ListItemAccordionProps>;
  ListItemButtonGroup: Partial<ButtonGroupProps>;
  ListItemCheckBox: Partial<CheckBoxProps>;
  ListItemContent: Partial<ViewProps>;
  ListItemChevron: Partial<IconProps>;
  ListItemInput: Partial<InputProps>;
  ListItemSubtitle: Partial<TextProps>;
  ListItemTitle: Partial<TextProps>;
  Overlay: Partial<OverlayProps>;
  PricingCard: Partial<PricingCardProps>;
  Rating: Partial<SwipeRatingProps>;
  AirbnbRating: Partial<TapRatingProps>;
  SearchBar: Partial<SearchBarProps>;
  Slider: Partial<SliderProps>;
  SocialIcon: Partial<SocialIconProps>;
  Text: Partial<TextProps>;
  Tile: Partial<TileProps>;
  Switch: Partial<SwitchProps>;
  Tooltip: Partial<TooltipProps>;
  colors: RecursivePartial<Colors>;
  Tab: Partial<TabProps>;
  TabItem: Partial<TabItemProps>;
  TabView: Partial<TabViewProps>;
  TabViewItem: Partial<ViewProps>;
  FAB: Partial<FABProps>;
  SpeedDial: Partial<SpeedDialProps>;
  SpeedDialAction: Partial<SpeedDialActionProps>;
  LinearProgress: Partial<LinearProgressProps>;
}

export type Theme<T = {}> = Partial<FullTheme> & T;

export type UpdateTheme = (updates: RecursivePartial<FullTheme>) => void;

export type ReplaceTheme = (updates: RecursivePartial<FullTheme>) => void;
