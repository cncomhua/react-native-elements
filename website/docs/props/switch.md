## Props

- [`value`](#value)
- [`disabled`](#disabled)
- [`color`](#color)
- [`theme`](#theme)
- [`onValueChange`](#onvaluechange)
- [`style`](#style)

---

## Reference

### `value`

The `value` of the Switch indicates the state on/off.

|  Type   | Default |
| :-----: | :-----: |
| boolean |  false  |

### `disabled`

The `disabled` disables the user interaction. (optional)

|  Type   | Default |
| :-----: | :-----: |
| boolean |  false  |

### `color`

The `color` of the Switch defines a color to the switch thumb. (optional)

|  Type  |    Default     |
| :----: | :------------: |
| string | Theme(Primary) |

### `theme`

The `theme` of the Switch provide a theme to the switch. (optional)

|  Type  |    Default     |
| :----: | :------------: |
| string | Theme(Primary) |

### `onValueChange`

The `onValueChange` function for container (optional)

|   Type   | Default |
| :------: | :-----: |
| function |  none   |

---

### `style`

The `style` is used to define user defined styles on the switch and allows styles provided by Native Switch.

|      Type      | Default |
| :------------: | :-----: |
| object (style) |  none   |
