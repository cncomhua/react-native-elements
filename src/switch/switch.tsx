import React from 'react';
import {
  Switch as NativeSwitch,
  SwitchProps as NativeSwitchProps,
  Platform,
} from 'react-native';
import { withTheme } from '../config';
import { Theme } from '../config/theme';

export type SwitchProps = NativeSwitchProps & {
  color?: string;
  theme?: Theme;
};

const Switch: React.FunctionComponent<SwitchProps> = ({
  value,
  disabled,
  onValueChange,
  color,
  style,
  theme,
  ...rest
}) => {
  // switchedOnColor deals with picking up a color provided as props by user or picks up default theme
  const switchedOnColor = color || theme.colors.primary;

  const onTintColor =
    Platform.OS === 'ios'
      ? switchedOnColor
      : !disabled
      ? switchedOnColor
      : theme.colors.disabled;

  const thumbTintColor =
    Platform.OS === 'ios'
      ? undefined
      : disabled
      ? theme.colors.disabled
      : value
      ? switchedOnColor
      : theme.colors.disabled;

  const props =
    Platform.OS === 'web'
      ? {
          activeTrackColor: onTintColor,
          thumbColor: thumbTintColor,
          activeThumbColor: switchedOnColor,
        }
      : {
          thumbColor: thumbTintColor,
          trackColor: {
            true: onTintColor,
            false: '',
          },
        };

  return (
    <NativeSwitch
      value={value}
      disabled={disabled}
      onValueChange={disabled ? undefined : onValueChange}
      style={style}
      {...props}
      {...rest}
    />
  );
};

export { Switch };
export default withTheme(Switch, 'Switch');
