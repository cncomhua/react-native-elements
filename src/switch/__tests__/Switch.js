import React from 'react';
import { Platform } from 'react-native';
import { Switch } from '../switch';
import theme from '../../config/theme';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Switch Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Switch value={true} color="green" />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render indeterminant without issues', () => {
    const component = shallow(<Switch value={true} color="red" />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render determinant without issues', () => {
    const component = shallow(
      <Switch value={false} disabled={true} color="green" />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with value', () => {
    const component = shallow(<Switch value={true} color={'purple'} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without value', () => {
    const component = shallow(<Switch color={'purple'} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with disabled', () => {
    const component = shallow(
      <Switch value={true} disabled={true} color={'purple'} />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without disabled', () => {
    const component = shallow(<Switch value={true} color={'purple'} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have an onValueChange event', () => {
    const onValueChange = jest.fn();
    const component = shallow(
      <Switch
        value={true}
        disabled={true}
        onValueChange={onValueChange}
        color={theme.colors.secondary}
      />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with style prop', () => {
    const component = shallow(
      <Switch value={true} color="green" style={{ margin: 10 }} />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders without style prop', () => {
    const component = shallow(<Switch value={true} theme={theme} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with props on web with disabled true', () => {
    Platform.OS === 'web';
    const component = shallow(
      <Switch
        disabled={true}
        value={false}
        color={'green'}
        trackColor={{ true: theme.colors.secondary, false: '' }}
      />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with props on web with disabled false', () => {
    Platform.OS === 'web';
    const component = shallow(
      <Switch
        disabled={false}
        value={false}
        color={'green'}
        trackColor={{ true: theme.colors.secondary, false: '' }}
      />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with props on ios', () => {
    Platform.OS === 'ios';
    const component = shallow(
      <Switch
        value={true}
        color="red"
        thumbTintColor={undefined}
        trackColor={{ true: theme.colors.secondary, false: '' }}
      />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders without props on ios', () => {
    Platform.OS === 'ios';
    const component = shallow(
      <Switch value={true} color="red" disabled={true} />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with disabled false props on ios', () => {
    const component = shallow(
      <Switch value={true} color="red" disabled={false} />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
