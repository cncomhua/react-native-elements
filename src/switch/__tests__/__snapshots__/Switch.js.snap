// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component renders with disabled false props on ios 1`] = `
<Switch
  disabled={false}
  trackColor={
    Object {
      "false": "",
      "true": "red",
    }
  }
  value={true}
/>
`;

exports[`Switch Component renders with props on ios 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "#ca71eb",
    }
  }
  value={true}
/>
`;

exports[`Switch Component renders with props on web with disabled false 1`] = `
<Switch
  disabled={false}
  trackColor={
    Object {
      "false": "",
      "true": "#ca71eb",
    }
  }
  value={false}
/>
`;

exports[`Switch Component renders with props on web with disabled true 1`] = `
<Switch
  disabled={true}
  trackColor={
    Object {
      "false": "",
      "true": "#ca71eb",
    }
  }
  value={false}
/>
`;

exports[`Switch Component renders with style prop 1`] = `
<Switch
  style={
    Object {
      "margin": 10,
    }
  }
  trackColor={
    Object {
      "false": "",
      "true": "green",
    }
  }
  value={true}
/>
`;

exports[`Switch Component renders without props on ios 1`] = `
<Switch
  disabled={true}
  trackColor={
    Object {
      "false": "",
      "true": "red",
    }
  }
  value={true}
/>
`;

exports[`Switch Component renders without style prop 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "#2089dc",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should have an onValueChange event 1`] = `
<Switch
  disabled={true}
  trackColor={
    Object {
      "false": "",
      "true": "#ca71eb",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render determinant without issues 1`] = `
<Switch
  disabled={true}
  trackColor={
    Object {
      "false": "",
      "true": "green",
    }
  }
  value={false}
/>
`;

exports[`Switch Component should render indeterminant without issues 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "red",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render with disabled 1`] = `
<Switch
  disabled={true}
  trackColor={
    Object {
      "false": "",
      "true": "purple",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render with value 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "purple",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render without disabled 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "purple",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render without issues 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "green",
    }
  }
  value={true}
/>
`;

exports[`Switch Component should render without value 1`] = `
<Switch
  trackColor={
    Object {
      "false": "",
      "true": "purple",
    }
  }
/>
`;
